book_manager.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'book'
+require_relative 'base_manager'
+
+class BookManager < BaseManager
+  attr_accessor :books
+
+  def initialize
+    super
+    @books = []
+  end
+
+  # List all books
+  def list_all_books
+    check_empty_list(@books, 'books')
+
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+  end
```

Comment:
- [ ] @NicolasAndrehh and @simplegoose great work in refactoring your code, however, I would suggest you move this function to another file, according to the `Single-responsibility Principle` each file should have a single responsibility this will help minimize errors debug quickly and also scale up the app easier in future, in our case all other methods handle inputs but this method does not.